src/source_sampling.cpp

```diff
@@ -394,3 +399,39 @@ int pyne::AliasTable::sample_pdf(double rand1, double rand2) {
   int i = (int) n * rand1;
   return rand2 < prob[i] ? i : alias[i];
 }
+
+pyne::SourceParticle::SourceParticle() {
+    this->x = -1.0;
+    this->y = -1.0;
+    this->z = -1.0;
+    this->e = -1.0;
+    this->w = -1.0;
+    this->c = -1;
+}
+
+pyne::SourceParticle::SourceParticle(double x, double y, double z,
```

Comment:
Better to use function arguments that are different names from member variables, and then
A. Avoid using `this` pointer, and
B. Use initialization lists